zeppelin-web/src/components/repository-create/repository-dialog.html

```diff
@@ -28,7 +28,7 @@ <h4 class="modal-title">Add New Repository</h4>
             <div class="form-group">
               <label class="control-label col-sm-2" for="repoId">ID</label>
               <div class="col-sm-10">
-                <input type="text" class="form-control" id="repoId" ng-model="newRepoSetting.id"
+                <input type="text" pattern="^[a-zA-Z0-9]*$" class="form-control" id="repoId" ng-model="newRepoSetting.id"
```

Comment:
If it's because of https://github.com/apache/zeppelin/blob/master/zeppelin-server/src/main/java/org/apache/zeppelin/rest/InterpreterRestApi.java#L274 `repository id` is path of the restAPI, other approach would be encode / decode `reponsitory id` before and after call REST API. Using `encodeURI(uri)` in js and `URLDecoder` in java